src/components/dock/dock.types.ts

```diff
@@ -0,0 +1,53 @@
+export interface DockItem {
+    /**
+     * A non-changing value to uniquely identify each item.
+     */
+    id: string;
+
+    /**
+     * Text to display for the item.
+     */
+    label: string;
+
+    /**
+     * Name of the icon to use.
+     */
+    icon: string;
+
+    /**
+     * Additional helper text for the button.
+     * Example usage can be a keyboard shortcut to activate the button.
+     */
+    helperLabel?: string;
+
+    /**
+     * Whether the button should indicate it is selected.
+     * The button should be selected when … (@Kiarokh: add description here please!)
```

Comment:
@Kiarokh Can you improve this description for me? I didn't really know what to write… 😄